airflow/models/dag.py

```diff
@@ -1504,6 +1504,8 @@ def filter_task_group(group, parent_group):
                 if isinstance(child, BaseOperator):
                     if child.task_id in dag.task_dict:
                         copied.children[child.task_id] = dag.task_dict[child.task_id]
+                    else:
+                        copied.used_group_ids.discard(child.task_id)
```

Comment:
Can you write a unit test for it please?

Reply:
I added another assert to an existing test of the `sub_dag` function, but the quarantined tests are failing (not sure if that's a problem)